onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```
@@ -221,13 +239,16 @@ struct OpenVINO_Provider : Provider {
     pi.device_type = ParseDeviceType(ov_core, provider_options, "device_type");
 
     if (provider_options.contains("device_id")) {
-      std::string dev_id = provider_options.at("device_id").data();
-      LOGS_DEFAULT(WARNING) << "[OpenVINO] The options 'device_id' is deprecated. "
-                            << "Upgrade to set deice_type and precision session options.\n";
-      if (dev_id == "CPU" || dev_id == "GPU" || dev_id == "NPU") {
-        pi.device_type = std::move(dev_id);
+      std::string luid_str = provider_options.at("device_id").data();
+      if (luid_str.empty()) {
+        ORT_THROW("[ERROR] [OpenVINO] Device id is empty");
       } else {
-        ORT_THROW("[ERROR] [OpenVINO] Unsupported device_id is selected. Select from available options.");
+        std::string device_name = map_LUID_to_device(luid_str);
+        if (device_name == "No_match_found") {
+          ORT_THROW(" [ERROR] [OpenVINO] Invalid LUID. No matching hardware is found");
+        } else {
+          pi.device_type = device_name;
```

What happens when user provides a  "device_type" and "device_id" that are mismatching?
For example, the user passes "device_type" as "CPU" and
"device_id" as the "LUID of a valid GPU"

---

Expected behavior should be, the LUID would be only checked for expected device_type. 
So, in the above example CPU LUID either would not exist or will not match with LUID provided in device_id. Execution should error out saying "Device_Id not found in device_type- CPU."